src/compiler/scanner.ts

```diff
@@ -1278,13 +1282,14 @@ export function createScanner(languageVersion: ScriptTarget,
      * Sets the current 'tokenValue' and returns a NoSubstitutionTemplateLiteral or
      * a literal component of a TemplateExpression.
      */
-    function scanTemplateAndSetTokenValue(isTaggedTemplate: boolean): SyntaxKind {
+    function scanTemplateAndSetTokenValue(shouldEmitInvalidEscapeError: boolean): SyntaxKind {
         const startedWithBacktick = text.charCodeAt(pos) === CharacterCodes.backtick;
 
         pos++;
         let start = pos;
         let contents = "";
         let resultingToken: SyntaxKind;
+        rangesOfOctalSequences = [];

```

Comment:
To avoid an allocation unless necessary, set this to `undefined`. Then every `rangesOfOctalSequences.push(...)` can become `(rangesOfOctalSequences ??= []).push(...)`.